docs/en/query_language/select.md

```diff
@@ -108,7 +108,7 @@ A table function may be specified instead of a table. For more information, see
 To execute a query, all the columns listed in the query are extracted from the appropriate table. Any columns not needed for the external query are thrown out of the subqueries.
 If a query does not list any columns (for example, SELECT count() FROM t), some column is extracted from the table anyway (the smallest one is preferred), in order to calculate the number of rows.
 
-The FINAL modifier can be used only for a SELECT from a CollapsingMergeTree table. When you specify FINAL, data is selected fully "collapsed". Keep in mind that using FINAL leads to a selection that includes columns related to the primary key, in addition to the columns specified in the SELECT. Additionally, the query will be executed in a single stream, and data will be merged during query execution. This means that when using FINAL, the query is processed more slowly. In most cases, you should avoid using FINAL. For more information, see the section "CollapsingMergeTree engine".
+The FINAL modifier can be used only for a SELECT from ReplacingMergeTree, SummingMergeTree, AggregatingMergeTree, CollapsingMergeTree and VersionedCollapsingMergeTree tables. When you specify FINAL, data is selected fully "merged". Keep in mind that using FINAL leads to a selection that includes columns related to the primary key, in addition to the columns specified in the SELECT. Additionally, the query will be executed in a single stream, and data will be merged during query execution. This means that when using FINAL, the query is processed more slowly. In most cases, you should avoid using FINAL.
```

Comment:
Did you forget GraphiteMergeTree?

PS. You can write - all MergeTree engines with non-trivial merge algorithm.

Reply:
I did not. I double checked.

SELECT *
FROM test
FINAL

Received exception from server (version 19.14.1):
Code: 181. DB::Exception: Received from localhost:9000. DB::Exception: Storage GraphiteMergeTree doesn't support FINAL.

CREATE TABLE test
(
    `Path` String,
    `Time` DateTime,
    `Value` Int64,
    `Version` Int64
)
ENGINE = GraphiteMergeTree('graphite_rollup_example')
ORDER BY tuple()

Reply:
I can write all  MergeTree engines except MergeTree and GraphiteMergeTree

Reply:
Ok. No need for it.